searx/engines/google_images.py

```diff
@@ -77,6 +77,22 @@ def scrap_out_thumbs(dom):
     return ret_val
 
 
+def scrap_img_by_id(script, data_id):
+    """Get full image URL by data-id in parent element
+    """
+    img_url = ''
+    _script = script.split('\n')
+    for i, line in enumerate(_script):
+        if 'gstatic.com/images' in line and data_id in line:
+            try:
+                url_line = _script[i + 1]
+                img_url = url_line.split('"')[1]
+                img_url = img_url.replace(r'\u00', '%')
```

Comment:
@gangural thanks for your PR / static code analysis is not easy, I find this solution very good.  There is only one point I would improve: This replacement does not worked in some examples I found.

I implemented a patch on top of your branch: https://github.com/return42/searx/commit/63bb1d65616e2ddfb4f45b9c0c0bd395370d8e4f

You find a example in the commit message. If you think my patch fits better, pull it from my branch: https://github.com/return42/searx/tree/google-img-fix

Side note: in my patch I removed the try/except block since I do not had any exception  .. Did you had any?

Reply:
I had exceptions while writing code. Now they are probably don't appear. About url-encoding, I thought that all web servers will interpret `%3d` as `=` but I was wrong.